api/apps/api/src/modules/authentication/password-recovery/mailer.ts

```diff
@@ -25,52 +34,68 @@ const passwordResetPrefixToken = Symbol('password reset prefix token');
 export const passwordResetPrefixProvider: FactoryProvider<string> = {
   provide: passwordResetPrefixToken,
   useFactory: () => {
-    const prefix = AppConfig.get<string>('passwordReset.tokenPrefix');
-    return prefix;
+    return AppConfig.get<string>('passwordReset.tokenPrefix');
   },
 };
 
+const signUpConfirmationPrefixToken = Symbol('sign-up confirm prefix token');
+export const signUpConfirmationPrefixProvider: FactoryProvider<string> = {
+  provide: signUpConfirmationPrefixToken,
+  useFactory: () => AppConfig.get<string>('signUpConfirmation.tokenPrefix'),
+};
+
 @Injectable()
 export class SparkPostMailer implements Mailer {
   private readonly logger = new Logger(this.constructor.name);
 
   constructor(
     private readonly sparkpost: Sparkpost,
+    @Inject(forwardRef(() => UsersService))
     private readonly usersService: UsersService,
     @Inject(passwordResetPrefixToken)
     private readonly passwordResetPrefix: string,
   ) {}
-  async sendPasswordChangedConfirmation(userId: string): Promise<void> {
-    const user = await this.usersService.getById(userId);
+
+  private async sendEmail(
+    template: string,
```

Comment:
this suggestion borders on (or may be firmly within the domain of) overengineering, but - given that we (and any admin of a self-hosted instance) are bound to manually sync template names between application and sparkpost, could it be prudent to narrow the typing of this parameter to enum values so that we can have one single point where we list all the templates (and can change them if ~needed~ wanted - the template names we have already cause my ocd to surface given their lack of consistency, even if this is practically irrelevant)?

Reply:
I think that makes perfect sense, thanks for the suggestion Andrea :bow: I wish we had done this in RW API for instance, it was a pain in the *** to find a list of all the templates being used across the API, so for sure this will be useful in the future :+1: 

I'll make the change :rocket: